packages/components/src/components/assay/AssayRunDeleteModal.tsx

```diff
@@ -0,0 +1,61 @@
+import React, { useState } from 'react';
+
+import { createDeleteErrorNotification, createDeleteSuccessNotification, deleteAssayRuns, Progress } from '../..';
+
+import { AssayRunDeleteConfirmModal } from './AssayRunDeleteConfirmModal';
+
+interface Props {
+    afterDelete: () => void;
+    afterDeleteFailure: () => void;
+    numToDelete: number;
+    onCancel: () => void;
+    selectionKey?: string;
+    selectedRowId?: string;
+}
+
+export function AssayRunDeleteModal(props: Props) {
+    const { afterDelete, afterDeleteFailure, numToDelete, onCancel, selectionKey, selectedRowId } = props;
+    const [showProgress, setShowProgress] = useState<boolean>();
+    const noun = numToDelete === 1 ? ' assay run' : ' assay runs';
+
+    function onConfirm(): void {
+        setShowProgress(true);
+        deleteAssayRuns(selectionKey, selectedRowId, true)
```

Comment:
nit: I'd recommend trying to move towards using `async/await` rather than promises `then/catch`.

Reply:
Good call I'll start trying to use async more